tools/template-check/gotemplate/gotemplate.go

```diff
@@ -12,18 +12,25 @@ import (
 // Note: this is allowlisted to combat other issues like `=` instead of `==`, but it is possible we
 // need to add more options to this list in the future, like `private`. The main thing we want to
 // prevent is targeting `beta` in version guards, because it mishandles either `ga` or `private`.
+
+// TODO: refactor to be a regex as the permitted formats are more varied under go templates
```

Comment:
That was my thought, maybe something like this?
`\{\{-? if (?:ne|eq) \$\.TargetVersionName [\x60\"]ga[\x60\"] \-?\}\}` (assuming regular go regex vs. re2)

(and could also choose to be flexible about spaces, though guessing better not here?)

https://regex101.com/r/jh3Uct/1

Reply:
Yeah, actually doing the expression wouldnt be hard, just need to adjust a bit of the lower logic (as currently it works by doing a literal string.contain check over the entire array of allowed guards)